### PR TITLE
Re-run make with verbose output on failure in CI script

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
This is working, if you look here: https://lc.llnl.gov/gitlab/radiuss/Umpire/-/jobs/126984#L501 the first parallel make fails, then it re-runs verbosely.

File: scripts/gitlab/build_and_test.sh [hunk 1]
```diff
@@ -106,7 +106,9 @@ then
     cmake \
       -C ${hostconfig_path} \
       ${project_dir}
-    cmake --build . -j
+    if ! cmake --build . -j; then
+      cmake --build . --verbose -j 1
```
Comment (mcfadden8) [inline scripts/gitlab/build_and_test.sh:110]: Looks good!

For clarity and being able to more easily search around in the log file, I would suggest:
```suggestion
      echo ""
      echo ""
      echo "Build failed, running build again sequentially in verbose mode"
      echo ""
      echo ""
      cmake --build . --verbose -j 1
```

Comment (davidbeckingsale) [reply scripts/gitlab/build_and_test.sh:113]: Done!

**Comment by davidbeckingsale** [comment]
LGTM

**Review by mcfadden8 — APPROVED** [review]

**Review by kab163 — APPROVED** [review]

**Pull request merged**